runtime-library/libdyntype/extref/extref.c

```diff
@@ -277,6 +277,62 @@ extref_invoke(dyn_ctx_t ctx, const char *name, dyn_value_t obj, int argc,
     return res;
 }
 
+dyn_value_t
+extref_get_keys(dyn_ctx_t ctx, dyn_value_t obj)
+{
+    dyn_value_t arr = NULL, str = NULL;
+    void *meta_addr = NULL;
+    uint32_t prop_count = 0, i = 0, iter_prop_count = 0;
+    char *prop_name = NULL;
+    char **prop_name_list = NULL;
+    EXTREF_PROLOGUE()
+
+    if (ext_tag == ExtObj) {
+        wasm_obj_t obj_struct =
+            (wasm_obj_t)wamr_utils_get_table_element(exec_env, table_index);
+        /* get meta, get prop names */
+        meta_addr = get_meta_of_object(exec_env, obj_struct);
+        prop_count = get_meta_fields_count(meta_addr);
+        if (prop_count > 0) {
+            prop_name_list = wasm_runtime_malloc(prop_count * sizeof(char *));
+        }
+        for (i = 0; i < prop_count; i++) {
+            prop_name = (char *)get_field_name_from_meta_index(
+                exec_env, meta_addr, FIELD, i);
+            if (prop_name) {
```

Comment:
What if prop_name == NULL?

Reply:
If `prop_count > 0`, the `prop_name` must not equal with NULL, maybe I can add a judgement here, if `prop_name == NULL`, then set an exception.

Reply:
If it's not possible to be NULL, should use `bh_assert(prop_name)`.
But seems it's possible here because we can't assume the meta is always correct

Reply:
If we get `prop_name` as null since meta error, seems that we should also set exception here?